dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -195,41 +217,31 @@ private DependencyNode resolveCompileTimeDependencies(
    * @param artifacts Maven artifacts to retrieve their dependencies
    * @return dependency graph representing the tree of Maven artifacts
    */
-  public DependencyGraphResult buildLinkageCheckDependencyGraph(List<Artifact> artifacts) {
+  public DependencyGraphResult buildFullDependencyGraph(List<Artifact> artifacts) {
     ImmutableList<DependencyNode> dependencyNodes =
         artifacts.stream().map(DefaultDependencyNode::new).collect(toImmutableList());
-    return buildDependencyGraph(
-        dependencyNodes, GraphTraversalOption.FULL_DEPENDENCY_WITH_PROVIDED);
+    return buildDependencyGraph(dependencyNodes, GraphTraversalOption.FULL);
   }
 
   /**
-   * Finds the full compile time, transitive dependency graph including duplicates and conflicting
-   * versions.
+   * Builds the transitive dependency graph as seen by Maven. It does not include duplicates and
+   * conflicting versions. That is, this resolves conflicting versions by picking the first version
+   * seen. This is how Maven normally operates.
    */
-  public DependencyGraphResult buildCompleteGraph(Dependency dependency) {
-    DefaultDependencyNode root = new DefaultDependencyNode(dependency);
-    return buildDependencyGraph(ImmutableList.of(root), GraphTraversalOption.FULL_DEPENDENCY);
-  }
-
-  /**
-   * Finds the complete transitive dependency graph as seen by Maven. It does not include duplicates
-   * and conflicting versions. That is, this resolves conflicting versions by picking the first
-   * version seen. This is how Maven normally operates.
-   */
-  public DependencyGraphResult buildGraph(Dependency dependency) {
+  public DependencyGraphResult buildMavenDependencyGraph(Artifact artifact) {
```

Comment:
Changed the argument type to Artifact because all caller was wrapping the argument with `new Dependency(artifact, "compile")`. The caller should not care about the root dependency's scope.

Reply:
No, I think we do want dependency here. Conceptually graphs are built from dependencies, not form artifacts.